IdentityCore/src/telemetry/request_telemetry/MSIDCurrentRequestTelemetry.m

```diff
@@ -29,17 +29,70 @@ @implementation MSIDCurrentRequestTelemetry
 
 - (NSString *)telemetryString
 {
-    return @"";
+    return [self serializeCurrentTelemetryString];
 }
 
-- (instancetype)initWithTelemetryString:(__unused NSString *)csvString error:(__unused NSError **)error
+- (instancetype)initWithTelemetryString:(NSString *)telemetryString error:(NSError **)error
 {
     self = [super init];
     if (self)
     {
+        NSError *internalError;
+        [self deserializeCurrentTelemetryString:telemetryString error:&internalError];
         
+        if (internalError) {
+            MSID_LOG_WITH_CTX_PII(MSIDLogLevelError, nil, @"Failed to initialize server telemetry, error: %@", MSID_PII_LOG_MASKABLE(internalError));
+            if (error) *error = internalError;
+            return nil;
+        }
     }
     return self;
 }
 
+#pragma mark - Private
+
+-(NSString *)serializeCurrentTelemetryString
+{
+    int forceRefreshValue = (self.forceRefresh ? 1 : 0);
+    NSString *telemetryString = [NSString stringWithFormat:@"%ld|%ld,%d|", self.schemaVersion, self.apiId, forceRefreshValue];
+    
+    // Make sure string to be returned is less than 4kB in size
+    if ([telemetryString lengthOfBytesUsingEncoding:NSUTF8StringEncoding] > 4000)
+    {
+        return nil;
```

Comment:
Do we want to send remaining telemetry below 4k?

Reply:
Yes, we will want to send all telemetry in the string under 4k once there are platform fields implemented as part of the current request telemetry. In this stage only the default fields will be sent, and since they are all specified by the schema to be of type int, this check is probably unnecessary.

Reply:
Makes total sense to me :) I was just curious. Resolving this comment.